### PR TITLE
Add customer to transactions table and filter

### PR DESCRIPTION
**Pull request by Gataniel**

### CODE REVIEW

File: app/views/transactions/_transactions_table.html.slim [hunk 1]
```diff
@@ -7,6 +7,7 @@
             td = sort_link(@q, :amount_cents, 'Amount')
             td = sort_link(@q, :category_name, 'Category')
             td = 'Account'
+            td = sort_link(@q, :customer_id)
```
Comment (shir) [inline app/views/transactions/_transactions_table.html.slim]: @Gataniel don't sort by id. Sort by customer name

File: spec/factories.rb [hunk 2]
```diff
@@ -87,7 +87,7 @@
 
   factory :transaction do
     bank_account
-    category
+    category_id { create(:category, organization: bank_account.organization).id }
```
Comment (shir) [inline spec/factories.rb]: @Gataniel don't use it such way. Use association. If you have some problems with it then ask me for help.

File: spec/factories.rb [hunk 3]
```diff
@@ -97,20 +97,23 @@
...
     comission       50
     comment         "comment"
-  end
 
-  trait :with_different_currencies do
-    bank_account_id { create(:bank_account, balance: 99999, currency: "USD").id }
-    reference_id    { create(:bank_account, currency: "RUB").id }
-    exchange_rate   0.5
+    trait :with_different_currencies do
+      bank_account_id { create(:bank_account, balance: 99999, currency: "USD").id }
```
Comment (shir) [inline spec/factories.rb]: @Gataniel same here.

File: spec/factories.rb [hunk 4]
```diff
@@ -97,25 +97,29 @@
     trait :expense do
       association :category, :expense
     end
+
+    trait :with_customer do
+      customer_id { create(:customer, organization: bank_account.organization).id }
```
Comment (shir) [inline spec/factories.rb]: @Gataniel fix it too.

File: spec/factories.rb [hunk 5]
```diff
@@ -97,25 +97,29 @@
...
+    end
   end
 
   factory :transfer do
-    bank_account_id { create(:bank_account, balance: 99999).id }
-    from_currency   { BankAccount.find(bank_account_id).currency }
-    reference_id    { create(:bank_account,
-      organization: BankAccount.find(bank_account_id).organization).id }
-    to_currency     { BankAccount.find(reference_id).currency}
+    bank_account_id { create(:bank_account, balance: 99999, currency: 'USD').id }
```
Comment (shir) [inline spec/factories.rb]: @Gataniel and this

File: spec/factories.rb [hunk 6]
```diff
@@ -97,25 +97,29 @@
...
+    reference_id    { |t| create(:bank_account,
```
Comment (shir) [inline spec/factories.rb]: @Gataniel and this

File: spec/factories.rb [hunk 7]
```diff
@@ -97,25 +97,29 @@
...
 
-  trait :with_different_currencies do
-    bank_account_id { create(:bank_account, balance: 99999, currency: "USD").id }
-    from_currency   { BankAccount.find(bank_account_id).currency }
-    reference_id    { create(:bank_account,
-      organization: BankAccount.find(bank_account_id).organization, currency: "RUB").id }
-    to_currency     { BankAccount.find(reference_id).currency}
-    exchange_rate   0.5
+    trait :with_different_currencies do
+      bank_account_id { create(:bank_account, balance: 99999, currency: 'USD').id }
```
Comment (shir) [inline spec/factories.rb]: @Gataniel and this

File: spec/factories.rb [hunk 8]
```diff
@@ -97,25 +97,29 @@
...
+      reference_id    { |t| create(:bank_account, currency: 'RUB',
```
Comment (shir) [inline spec/factories.rb]: @Gataniel and this

File: spec/factories.rb (hunk 7 above)
Comment (Gataniel) [inline spec/factories.rb]: @shir I think its impossible, because `Transfer` is not Non ActiveRecord model

Comment (shir) [inline spec/factories.rb]: @Gataniel ahh...ok

**Pull request merged**